burn-wgpu/src/codegen/function.rs

```diff
@@ -22,14 +24,62 @@ impl Display for Function {
     }
 }
 
-fn format_powf(f: &mut core::fmt::Formatter<'_>, item: &Item) -> core::fmt::Result {
-    let elem = item.elem();
+fn format_powf_scalar(f: &mut core::fmt::Formatter<'_>, item: &Item) -> core::fmt::Result {
+    base_powf_fmt(f, item)?;
+
+    match item {
+        Item::Vec4(elem) => f.write_fmt(format_args!(
+            "
+fn powf(lhs: {item}, rhs: {elem}) -> {item} {{
+    return vec4(
+        powf_scalar(lhs[0], rhs),
+        powf_scalar(lhs[1], rhs),
```

Comment:
This is going to cause a problem, since pow scalar and pow elemwise can be in the same kernel with operation fusion. We probably need two different function names.

Reply:
I think I would rename `powf_scalar` to `powf_primitive` so that we can use `powf_scalar` for the "real" `powf_scalar` that can be vectorized.